### PR TITLE
Miscellaneous cleanup

### PR DESCRIPTION
**Pull request by basil**
Another bunch of miscellaneous cleanup hunks that I've been collecting in a branch for a few weeks now. I tested these in BOM with `PLUGINS=text-finder bash local-test.sh`.

### CODE REVIEW

File: src/main/java/org/jenkins/tools/test/PluginCompatTester.java [hunk 1]
```diff
@@ -488,7 +488,7 @@ public static void cloneFromScm(
      *   <li><code>git checkout FETCH_HEAD</code>
      * </ul>
      *
-     * @param gitURL The git native URL, see the <a
+     * @param gitUrl The git native URL, see the <a
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/PluginCompatTester.java:491]: This was not matching the case of the actual parameter.

File: src/main/java/org/jenkins/tools/test/hook/NodeCleanupBeforeCompileHook.java [hunk 2]
```diff
@@ -26,11 +26,10 @@ public Map<String, Object> action(Map<String, Object> moreInfo) {
             File pluginDir = (File) moreInfo.get("pluginDir");
             LOGGER.log(Level.INFO, "Executing node and node_modules cleanup hook");
             compile(pluginDir);
-            return moreInfo;
         } else {
             LOGGER.log(Level.INFO, "Hook not triggered; continuing");
-            return moreInfo;
         }
+        return moreInfo;
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/hook/NodeCleanupBeforeCompileHook.java:32]: One-line code deduplication.

File: src/main/java/org/jenkins/tools/test/model/hook/PluginCompatTesterHooks.java [hunk 3]
```diff
@@ -200,11 +199,8 @@ private Map<String, Queue<PluginCompatTesterHook>> findHooks(String stage) {
                         allForType.add(hook);
                         sortedHooks.put(plugin, allForType);
                     }
-                } catch (Exception ex) {
-                    LOGGER.log(
-                            Level.WARNING,
-                            "Error when loading " + c.getName() + "; continuing",
-                            ex);
+                } catch (ReflectiveOperationException e) {
+                    throw new RuntimeException("Error when loading " + c.getName(), e);
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/hook/PluginCompatTesterHooks.java:203]: This should never happen, and if it does it is a serious error that would invalidate our confidence in the test run, so throw it up and halt execution.

File: src/main/java/org/jenkins/tools/test/model/MavenPom.java [hunk 4]
```diff
@@ -229,13 +230,13 @@ public void addDependencies(
...
             List<String> toConvert)
-            throws IOException {
+            throws PomTransformationException {
         File pom = new File(rootDir.getAbsolutePath() + "/" + pomFileName);
         Document doc;
         try {
             doc = new SAXReader().read(pom);
         } catch (DocumentException x) {
-            throw new IOException(x);
+            throw new PomTransformationException("Failed to parse pom.xml", x);
```
Comment (basil) [inline src/main/java/org/jenkins/tools/test/model/MavenPom.java:239]: Making this class consistent with the error handling philosophy elsewhere of failing fast and avoiding abuse of `IOException`.

**Review by imonteroperez — APPROVED** [review]

**Review by jtnord — APPROVED** [review]

**Pull request merged**